ethereum/clpps/CLPPSToken.sol

```diff
@@ -0,0 +1,113 @@
+pragma solidity >=0.4.22 <0.7.0;
+
+library SafeMath {
+  function sub(uint256 a, uint256 b) internal pure returns (uint256) {
+    assert(b <= a);
+    return a - b;
+  }
+
+  function add(uint256 a, uint256 b) internal pure returns (uint256) {
+    uint256 c = a + b;
+    assert(c >= a);
+    return c;
+  }
+}
+
+contract CLPPSToken {
+  string public constant name = "CLPPSToken";
+  string public constant symbol = "CLPPS";
+  uint8 public constant decimals = 18;
+  address payable public clppsAdmin; // Address with privileges to mint new tokens
+
+  constructor(uint256 total, address payable admin) public {
+    clppsAdmin = admin;
+    totalSupply_ = total;
+    balances[msg.sender] = totalSupply_;
+  }
+
+  /****
+  * ERC-20 Compliant Interface
+  ****/
+  event Approval(address indexed tokenOwner, address indexed spender, uint tokens);
+  event Transfer(address indexed from, address indexed to, uint tokens);
+
+  mapping(address => uint256) balances;
+
+  mapping(address => mapping (address => uint256)) allowed;
+
+  uint256 totalSupply_;
+
+  using SafeMath for uint256;
+
+  function totalSupply() public view returns (uint256) {
+	  return totalSupply_;
+  }
+
+  function balanceOf(address tokenOwner) public view returns (uint) {
+    return balances[tokenOwner];
+  }
+
+  function transfer(address receiver, uint numTokens) public returns (bool) {
+    require(numTokens <= balances[msg.sender], "Insufficient Tokens");
+    balances[msg.sender] = balances[msg.sender].sub(numTokens);
+    balances[receiver] = balances[receiver].add(numTokens);
+    emit Transfer(msg.sender, receiver, numTokens);
+    return true;
+  }
+
+  function approve(address delegate, uint numTokens) public returns (bool) {
```

Comment:
This function seems to be potentially exploitable https://docs.google.com/document/d/1YLPtQxZu1UAvO9cZ1O2RPXBbT0mooh4DYKjA_jp-RLM/edit

I wonder if we should ensure to always change to 0 first?

Reply:
It may also make sense to check if `delegate` is not null

Reply:
good call!  i believe checking for address byte length is also a security recommendation.  i will add these!